interfaces/builtin/tee.go

```diff
@@ -0,0 +1,55 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const teeSummary = `allows access to Trusted Execution Environment devices`
+
+const teeBaseDeclarationSlots = `
+  tee:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+const teeConnectedPlugAppArmor = `
+# Description: for those who need to talk to the TEE subsystem over
+# /dev/tee[0-9]* and/or /dev/teepriv[0-0]*
+
+/dev/tee[0-9]* rw,
+/dev/teepriv[0-9]* rw,
+`
+
+var teeConnectedPlugUDev = []string{
+	`KERNEL=="tee[0-9]*"`,
+	`KERNEL=="teepriv[0-9]*"`,
+}
+
+func init() {
+	registerIface(&commonInterface{
+		name:                  "tee",
+		summary:               teeSummary,
+		implicitOnCore:        true,
+		implicitOnClassic:     true,
```

Comment:
Do we need this for classic? I can imagine it is useful for some specific context but I was wondering if this is by design or by accident.

Reply:
This is for a customer that wants to maintain the user space apps (tee-supplicant, tee-secure-watchdog) in snap packages in a brand store. The system using this brand store will be offered with classic as well as with core images, the configuration/combination is optionally chooseable by the end-customer buying the device. 

To not force the customer to maintain separate deb packages in a PPA for these userspace apps (the expectation is that all device related software will only be available as snaps), the interface needs to be available on core as well as on classic.

Reply:
hardening done ...